layout/responsive.go

```diff
@@ -0,0 +1,288 @@
+		ww := ResponsiveBreakpoint(windowSize.Width)
+		if ww <= SMALL {
+			size.Width = conf[SMALL] * containerSize.Width
+		} else if ww <= MEDIUM {
+			size.Width = conf[MEDIUM] * containerSize.Width
+		} else if ww <= LARGE {
+			size.Width = conf[LARGE] * containerSize.Width
+		} else {
+			size.Width = conf[XLARGE] * containerSize.Width
+		}
+
+		// place and resize the element
+		o.Move(pos)
+		o.Resize(size)
+
+		// to know where to place the next line
+		maxHeight = resp.maxFloat32(maxHeight, size.Height)
+
+		// move the next element to the right
+		pos.X += size.Width + theme.Padding()
+
+		// manage end of line, the next position overflows, so go to next line.
+		if pos.X >= containerSize.Width {
+			pos.X = 0
+			pos.Y += maxHeight
+			maxHeight = 0
+		}
+	}
+}
+
+// MinSize return the minimum size ot the layout.
+//
+// Implements: fyne.Layout
+func (resp *ResponsiveLayout) MinSize(objects []fyne.CanvasObject) fyne.Size {
+	if len(objects) == 0 {
+		return fyne.NewSize(0, 0)
+	}
+
+	var h, w, maxHeight float32
+	currentY := objects[0].Position().Y
+
+	for _, o := range objects {
+		if o == nil || !o.Visible() {
+			continue
+		}
+		if o.Position().Y != currentY {
+			currentY = o.Position().Y
+			// new line, so we can add the maxHeight to h
+			h += maxHeight
+
+			// drop the line
+			maxHeight = 0
+		}
+		maxHeight = resp.maxFloat32(maxHeight, o.MinSize().Height)
+	}
+	h += maxHeight + theme.Padding()
+
+	return fyne.NewSize(w, h)
+}
+
+// SetResponsiveConfig sets the configuration for the object.
+// It creates a new configuration if the object was not registered.
+func (resp *ResponsiveLayout) SetResponsiveConfig(object fyne.CanvasObject, conf ResponsiveConfiguration) {
+	resp.mutex.Lock()
+	defer resp.mutex.Unlock()
+
+	// drom from global
+	delete(responsiveConfigurations, object)
+
+	resp.configuration[object] = conf
+}
+
+// math.Max only works with float64, so let's make our own
+func (resp *ResponsiveLayout) maxFloat32(a, b float32) float32 {
+	return float32(math.Max(float64(a), float64(b)))
+}
+
+// NewResponsiveLayout return a responsive layout that will adapt objects with the responsive rules. To
+// configure the rule, each object could be encapsulated by a "Responsive" object.
+//
+// Example:
+//      container := NewResponsiveLayout(
+//          Responsive(label, 1, .5, .25),  // 100% for small, 50% for medium, 25% for large
+//          Responsive(button, 1, .5, .25), // ...
+//          label2,                         // this will be placed and resized with default behaviors
+//                                          // => 1, 1, 1
+//      )
+func NewResponsiveLayout(o ...fyne.CanvasObject) *fyne.Container {
+
+	r := &ResponsiveLayout{
+		configuration: make(map[fyne.CanvasObject]ResponsiveConfiguration),
+		mutex:         &sync.Mutex{},
+	}
+
+	for _, ob := range o {
+		if conf, ok := responsiveConfigurations[ob]; !ok {
```

Comment:
It would be good to hold a lock here to protect the access to responsiveConfigurations.

Reply:
fixed in https://github.com/fyne-io/fyne-x/pull/36/commits/fef4cbaa692e760d40d9c73ced3c6cb019725e7f

Reply:
fixed in https://github.com/fyne-io/fyne-x/pull/36/commits/01cfc672cf3af05914a4c0bcbfb604bdab72bf51